fiona/ogrext.pyx

```diff
@@ -255,18 +255,26 @@ cdef class FeatureBuilder:
         cdef void *cogr_geometry = NULL
 
         if not ignore_geometry:
-            cogr_geometry = OGR_F_GetGeometryRef(feature)
+            """ We steal the geometry: the geometry of the in-memory feature is now null
+            and we are responsible for cogr_geometry.
+            """
+            cogr_geometry = OGR_F_StealGeometry(feature)
 
             if cogr_geometry is not NULL:
-                code = OGR_G_GetGeometryType(cogr_geometry)
 
-                if 7 < code < 100:  # Curves.
+                code = base_geometry_type_code(OGR_G_GetGeometryType(cogr_geometry))
```

Comment:
Is this acceptable or would it be better style to create a separate variable for OGR_G_GetGeometryType(cogr_geometry)?

Reply:
This is fine @rbuffat .